client/landing/stepper/hooks/use-is-site-big-sky-eligible.ts

```diff
@@ -33,11 +35,25 @@ export function useIsBigSkyEligible( flowName?: string ) {
 
 	const isEligibleGoals = isGoalsBigSkyEligible( goals );
 	const isEligiblePlan = isPremiumPlan( product_slug ) || isBusinessPlan( product_slug );
+	const siteHasAiAssistantFeature =
+		site?.plan?.features?.active?.includes( FEATURE_AI_ASSISTANT ) ?? false;
 
 	if ( flowName === AI_SITE_BUILDER_FLOW ) {
 		return { isEligible: true };
 	}
 
+	if ( flowName === SITE_SETUP_FLOW ) {
+		return {
+			isEligible:
+				featureFlagEnabled &&
+				featurePostCheckoutAiStepEnabled &&
+				!! isOwner &&
+				siteHasAiAssistantFeature &&
```

Comment:
This check seems to be sufficient.
Sites with WPCOM_PERSONAL_AND_HIGHER_PLANS and Jetpack Sites which paid for this feature, will have it enabled.